kore/test/Test/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -933,8 +939,63 @@ test_andTermsSimplification =
                 testSet
             assertEqualWithExplanation "" (Just expected) actual
         ]
+    , testGroup "alias expansion"
+        [ testCase "alias() vs top" $ do
+            let
+                x = mkVariable "x"
+                alias = mkAlias' "alias1" x mkTop_
+                left = applyAlias' alias $ mkTop Mock.testSort
+            actual <- simplifyUnify left mkTop_
+            assertExpectTop actual
+        , testCase "alias1() vs alias2()" $ do
+            let
+                x = mkVariable "x"
+                leftAlias = mkAlias' "leftAlias" x mkTop_
+                left = applyAlias' leftAlias $ mkTop Mock.testSort
+                rightAlias = mkAlias' "rightAlias" x mkTop_
+                right = applyAlias' rightAlias $ mkTop Mock.testSort
+            actual <- simplifyUnify left right
+            assertExpectTop actual
+        , testCase "alias1(alias2()) vs top" $ do
+            let
+                x = mkVariable "x"
+                y = mkVariable "y"
+                alias1 = mkAlias' "alias1" x mkTop_
+                alias1App = applyAlias' alias1 $ mkSetVar (SetVariable y)
+                alias2 = mkAlias' "alias2" x alias1App
+                alias2App = applyAlias' alias2 $ mkTop Mock.testSort
+            actual <- simplifyUnify alias2App mkTop_
+            assertExpectTop actual
+        ]
     ]
 
+mkVariable :: Text -> Variable
+mkVariable ident = Variable (testId ident) Nothing Mock.testSort
+
+mkAlias' :: Text -> Variable -> TermLike Variable -> SentenceAlias (TermLike Variable)
+mkAlias' ident var inner =
+    mkAlias_ (testId ident) Mock.testSort [SetVar $ SetVariable var] inner
+
+applyAlias'
+    :: Ord variable
+    => SortedVariable variable
+    => Unparse variable
+    => SentenceAlias (TermLike Variable)
+    -> TermLike variable
+    -> TermLike variable
+applyAlias' alias arg = applyAlias alias [] [arg]
+
+assertExpectTop :: ([Pattern Variable], Maybe [Pattern Variable]) -> IO ()
+assertExpectTop =
+    assertEqualWithExplanation "" ([expectTop_], Just [expectTop_])
+  where
+    expectTop_ =
```

Comment:
Why not [`Kore.Internal.Pattern.top`](https://github.com/kframework/kore/blob/master/kore/src/Kore/Internal/Pattern.hs#L162)?